src/wallet/api/wallet.cpp

```diff
@@ -733,10 +733,10 @@ bool WalletImpl::close(bool store)
 
 std::string WalletImpl::seed() const
 {
-    std::string seed;
+    epee::wipeable_string seed;
     if (m_wallet)
         m_wallet->get_seed(seed);
-    return seed;
+    return std::string(seed.data(), seed.size()); // TODO
```

Comment:
This form of conversion to `std::string` shows up repetitively, so maybe add a member function `std::string wipeable_string::to_string() const`?

Reply:
I'd rather not, because it'd make it easy to leak the "secret" data.

Reply:
I can see eight instances of this form.

Reply:
This is too many :) Can you list the ones you found please ?

Reply:
language_base.h:
- L123: `MWARNING("Duplicate prefix in " << language_name << " word list: " << std::string(trimmed.data(), trimmed.size()));`
- L125: `throw std::runtime_error("Duplicate prefix in " + language_name + " word list: " + std::string(trimmed.data(), trimmed.size()));`

simplewallet.cpp:
- L2884: `multisig_keys = m_wallet->decrypt<epee::wipeable_string>(std::string(multisig_keys.data(), multisig_keys.size()), key, true);`

api/wallet.cpp:
- L739: `return std::string(seed.data(), seed.size()); // TODO`

wallet_rpc_server.cpp:
-L1522: `res.key = std::string(seed.data(), seed.size()); // send to the network, then wipe RAM :D`

unit_tests/mnemonics.cpp:
- L88: `seed = std::string(w_seed.data(), w_seed.size());`
- L111: `return_seed = std::string(w_return_seed.data(), w_return_seed.size());`
- L136: `return_seed = std::string(w_return_seed.data(), w_return_seed.size());`

Reply:
Thanks. Those are all OK, except for the one I marked TODO, as the GUI might not be able to keep its data wiped.

Reply:
OK. BTW, I'd like to try and see if I can easily activate this encryption feature for the GUI (since most novice users will be using the GUI).